sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java

```diff
@@ -31,6 +35,23 @@ void setUp() {
     GlobalEventEmitterProvider.resetForTest();
   }
 
+  @SuppressWarnings("ResultOfMethodCallIgnored")
+  @Test
+  void build_addMetricReaderCustomizerPrometheus() {
+    AutoConfiguredOpenTelemetrySdkBuilder builder = AutoConfiguredOpenTelemetrySdk.builder();
+
+    builder.addPropertiesSupplier(() -> singletonMap("otel.metrics.exporter", "prometheus"));
+    builder.addMetricReaderCustomizer(
+        (reader, config) -> {
+          assertThat(reader).isInstanceOf(PrometheusHttpServer.class);
+          return PrometheusHttpServer.builder().setPort(1234).build();
```

Comment:
Can you find a random unused port like we do in this [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/MetricReaderFactoryTest.java#L228-L237)?

Reply:
The tests run on a remote with a predictable environment and where port collisions are unlikely, but they also run locally where free ports cannot be guaranteed.